src/GovUk.Education.ExploreEducationStatistics.Admin/Services/ReleaseApprovalService.cs

```diff
@@ -188,6 +195,110 @@ public class ReleaseApprovalService : IReleaseApprovalService
+            // PublishStagedReleaseContent (2) - Runs after (1) and completes publishing of releases.
+
+            // The cron expressions are configurable per environment to allow different schedules for testing.
+
+            // There's a requirement that (1) and (2) always run at specific times in the Prod environment,
+            // e.g. (1) runs at 00:00:00 and (2) runs at 09:30:00 irrespective of daylight saving time.
+            // To avoid needing to adjust a cron expression for daylight saving time twice a year to express
+            // the desired time in UTC (e.g. changing it from '0 30 9 * * *' to '0 30 8 * * *' in British Summer Time),
+            // the functions have been configured to run in the UK timezone rather than the UTC default.
+
+            // Check if the functions have a scheduled occurrence on the publish date in the range between midnight
+            // (inclusive) and midnight the following day (exclusive), UK time.
+
+            var ukTimeZone = DateTimeExtensions.GetUkTimeZone();
+            var fromUtc = TimeZoneInfo.ConvertTimeToUtc(publishDate.Date, ukTimeZone);
+            var toUtc = fromUtc.AddDays(1).AddTicks(-1);
+
+            // The publish date cannot be scheduled if it's already passed
+            if (_dateTimeProvider.UtcNow > toUtc)
+            {
+                return false;
+            }
+
+            // The range should begin now rather than at midnight if the publish date is today
+            if (_dateTimeProvider.UtcNow > fromUtc)
+            {
+                fromUtc = _dateTimeProvider.UtcNow; 
+            }
+
+            // Publishing won't occur unless there's an occurrence of (1) between the publishing range
+            if (TryGetNextOccurrenceForCronExpression(
+                    cronExpression: _options.PublishReleasesCronSchedule,
+                    fromUtc: fromUtc,
+                    toUtc: toUtc,
+                    timeZoneInfo: ukTimeZone,
+                    fromInclusive: true,
+                    toInclusive: true,
+                    out var nextOccurrenceUtc))
+            {
+                // Publishing won't occur unless there's an occurrence of (2) after (1) but before the end of the range
+                return TryGetNextOccurrenceForCronExpression(
+                    cronExpression: _options.PublishReleaseContentCronSchedule,
+                    fromUtc: nextOccurrenceUtc.Value,
+                    toUtc: toUtc,
+                    timeZoneInfo: ukTimeZone,
+                    fromInclusive: true,
+                    toInclusive: true,
+                    out _);
+            }
+
+            return false;
+        }
+
+        private static bool TryGetNextOccurrenceForCronExpression(string cronExpression,
+            DateTime fromUtc,
+            DateTime toUtc,
+            TimeZoneInfo timeZoneInfo,
+            bool fromInclusive,
+            bool toInclusive,
```

Comment:
We can yes but then the `out` parameter isn't last which I think looks a bit strange:

```
TryGetNextOccurrenceForCronExpression(string cronExpression,
            DateTime fromUtc,
            DateTime toUtc,
            TimeZoneInfo timeZoneInfo,
            [NotNullWhen(true)] out DateTime? nextOccurrenceUtc,
            bool fromInclusive = true,
            bool toInclusive = true)
```

I thought the use of the `TryGet` pattern here was pretty helpful, because rather than something like this

```
var nextOccurrenceUtc = GetNextOccurrenceForCronExpression(...);

if (nextOccurrenceUtc.HasValue)
{
    return GetNextOccurrenceForCronExpression(...).HasValue;
}
```

we can simply do

```
if (TryGetNextOccurrenceForCronExpression(..., out var nextOccurrenceUtc))
{
    return TryGetNextOccurrenceForCronExpression(..., out _);
}
```

Not sure what you think, but with the default parameters I think it's better now to not use TryGet pattern. I've switched it up in 5684bb144953007657117420068a36d5b673c94f.